validphys2/src/validphys/core.py

```diff
@@ -266,83 +264,153 @@ def get_kinlabel_key(process_label):
                          "variables matching  the process %s Check the "
                          "labels defined in commondata.cc. " % (l)) from e
 
-CommonDataMetadata = namedtuple('CommonDataMetadata', ('name', 'nsys', 'ndata', 'process_type'))
 
-def peek_commondata_metadata(commondatafilename):
-    """Check some basic properties commondata object without going though the
-    trouble of processing it on the C++ side"""
-    with open(commondatafilename) as f:
-        try:
-            l = f.readline()
-            name, nsys_str, ndata_str = l.split()
-            l = f.readline()
-            process_type_str = l.split()[1]
-        except Exception:
-            log.error(f"Error processing {commondatafilename}")
-            raise
+class CommonDataSpec(TupleComp):
 
-    return CommonDataMetadata(name, int(nsys_str), int(ndata_str),
-                              get_kinlabel_key(process_type_str))
+    def __init__(self, name, variant, metadata_file=None):
+        parent = metadata_file.parent
+        metadata = yaml.safe_load(metadata_file.open("r", encoding="utf-8"))
+
+        # Prepare to read the defaults
+        dat = metadata["data_central"]
+        unc = metadata["data_uncertainties"]
+        kin = metadata["data_kinematics"]
+
+        # Check whether the variants override any of the files
+        # TODO: various try-except in case things don't work as expected
+        if variant is None:
+            variant_info = metadata[variant]
+            dat = variant_info.get("data", dat)
+            unc = variant_info.get("uncertainties", unc)
+            kin = variant_info.get("kinematics", kin)
+
+        self._data_file = parent / dat
+        self._kinematics_file = parent / kin
+        self._uncertainties_file = parent / unc
+        # Check files exist and all that
+
+        # Public attributes
+        self.name = name
+        self.variant = variant
+        self.metadata = metadata
 
+        # lazy loading
+        self._data = None
+        self._uncertainties = None
+        self._kinematics = None
 
-class CommonDataSpec(TupleComp):
-    def __init__(self, datafile, sysfile, plotfiles, name=None, metadata=None):
-        self.datafile = datafile
-        self.sysfile = sysfile
-        self.plotfiles = tuple(plotfiles)
-        self._name=name
-        self._metadata = metadata
-        super().__init__(datafile, sysfile, self.plotfiles)
+        super().__init__(name, variant)
 
     @property
-    def name(self):
-        return self.metadata.name
+    def data(self):
+        """Reads the data as a 1-d numpy array"""
+        if self._data is None:
+            datayaml = yaml.safe_load(self._data_file.read_text(encoding="utf-8"))
+            self._data = pd.DataFrame.from_records(datayaml["data_central"], index="index")
+            self._data.rename(columns={self._data.columns[0]:"data"}, inplace=True)
+        return self._data
+
+    @property
```

Comment:
`validphys` at least declares support for [3.6 on](https://github.com/NNPDF/nnpdf/blob/7ceadf8259bc361d0a86fcc542ff22b3b454e72d/validphys2/setup.py#L5-L7), while `cached_property` has been introduced in 3.8.
I have no strong opinion, but the two things together are inconsistent, so you can stop supporting 3.6 and 3.7 (3.6 is barely used at this point, and even 3.7 has limited user base) and write it in `validphys`, or you have to give up on `cached_property`.

Reply:
Effectively we support whatever we build packages for, which is 3.8 and 3.9. PRs to specify this in other places would be welcome.

Reply:
Yeah, we decided that anything not in conda was not supported. But since I use 3.10 I'm always wary of introducing new stuff :_ 
I should also introduce also the walrus operator everywhere, I really like it and thought it was 3.9 but it's 8!